PyEMD/significancetest.py

```diff
@@ -0,0 +1,127 @@
+from PyEMD import EMD, EEMD, CEEMDAN
+import numpy as np
+from scipy import stats
+import math
+
+
+#helper function: normalize data
+def normalize(data):
+    mean = sum(data)/len(data)
+    dmin = min(data)
+    dmax = max(data)
+    ret = np.array([])
+    for val in data:
+        ret = np.append(ret, (val-mean)/(dmax-dmin))
+    return ret
+
+#helper function: find zero-crossings in IMF, used for finding mean period of IMF
+def sign_change(data):
+    l = len(data)
+    counter=0
+    for x in range(l-1):
+        if (data[x]>0 or data[x] == 0) and data[x+1] < 0:
+            counter+=1
+    return counter
```

Comment:
Is that a general method for checking sign change? Based on code it seems to only check when going from positive (data[x] >= 0) to negative (data[x+1] < 0). What about from negative (data[x] <= 0) to positive (data[x+1] > 0)?

Code-wise, unfortunately, Python's `for` loops are rather slow. Since you have data in numpy it is advised to do a high level operation. To check whether there's a zero crossing, likely the best is to check whether there's a negative sign when multiplying consecutive values, e.g.
```python
def sign_change(data):
    prev_values = data[:-1]
    next_values = data[1:]
    counter_idx = (prev_values * next_values) < 0
    return np.sum(counter_idx)
```

Here. `np.sum` works because implicitly `False` and `True` are evaluated to `0` and `1`, respectively.

Reply:
The `sign_change` function was used to find mean period of IMF. However, I just found a function in scipy which provides an easier solution, and is probably better computationally. Please see if this will be a better solution,
```python
def mean_period(data):
    return len(data)/len(argrelextrema(data, np.greater)[0])
```
The code bassically finds the number of peaks(local extrema) in the signal and divides it from the length of the time-series to estimate the mean period. The same method was also employed by Wu and Huang in their paper.

Reply:
Sweet!